x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/dataframe/process/AnalyticsResultProcessor.java

```diff
@@ -255,14 +259,36 @@ private InferenceConfig buildInferenceConfig(TargetType targetType) {
                     .setNumTopFeatureImportanceValues(regression.getBoostedTreeParams().getNumTopFeatureImportanceValues())
                     .build();
             default:
-                setAndReportFailure(ExceptionsHelper.serverError(
+                throw ExceptionsHelper.serverError(
                     "process created a model with an unsupported target type [{}]",
                     null,
-                    targetType));
-                return null;
+                    targetType);
         }
     }
 
+    PredictedFieldType getPredictedFieldType(Classification classification) {
```

Comment:
One suggestion that would help keeping the strings "string", "int", "bool" in one class would be as follows:
1. Make `Classification.getPredictionFieldType` method return the new `PredictedFieldType` enum.
2. Here (i.e. in `AnalyticsResultProcessor`) remove the translation in lines 277-285 (maybe keep the null -> STRING translation)
3. There (i.e. in `Classification`) in `getParams` method translate the enum to appropriate string ("string", "int", "bool").